frontend/src/metabase/timelines/common/components/NewTimelineModal/NewTimelineModal.tsx

```diff
@@ -43,16 +38,28 @@ const NewTimelineModal = ({
     <div>
       <ModalHeader title={t`New event timeline`} onClose={onClose} />
       <ModalBody>
-        <Form
-          form={forms.details}
+        <TimelineForm
           initialValues={initialValues}
-          isModal={true}
           onSubmit={handleSubmit}
-          onClose={onCancel}
+          onCancel={onCancel}
         />
       </ModalBody>
     </div>
   );
 };
 
+const getInitialValues = (collection: Collection): TimelineData => ({
+  name: "",
+  description: "",
+  collection_id: canonicalCollectionId(collection.id),
+  icon: getDefaultTimelineIcon(),
+  default: false,
+  archived: false,
+});
+
+const getSubmitValues = (values: TimelineData): TimelineData => ({
+  ...values,
+  description: values.description || null,
```

Comment:
Just curious. We didn't set the description to `null` when it's empty e.g. when editing values and deleting everything in the description input box. Is it a practice/convention that we should set the input value to `null` when we clear the input value?

Reply:
Yeah, that's the convention used by many API handlers. It's not universal through.